pcsx2/gui/Panels/MemoryCardListPanel.cpp

```diff
@@ -784,26 +775,40 @@ bool Panels::MemoryCardListPanel_Simple::UiDuplicateCard(McdSlotItem& src, McdSl
 		wxString errMsg;
 		if (!isValidNewFilename(newFilename, basepath, errMsg, 5))
 		{
-			wxString message;
-			message.Printf(_("Failed: %s"), WX_STR(errMsg));
-			Msgbox::Alert(message, _("Duplicate memory card"));
-			continue;
-		}
-
-		dest.Filename = newFilename;
-		break;
-	}
+			wxString newFilename=L"";
+			newFilename = wxGetTextFromUser(_("Select a name for the duplicate\n( '.ps2' will be added automatically)"), _("Duplicate memory card"));
+			if( newFilename==L"" )
+			{
+				//Msgbox::Alert( _("Duplicate canceled"), _("Duplicate memory card") );
+				return false;
+			}
+			newFilename += L".ps2";
 
-	wxFileName srcfile(basepath + src.Filename);
-	wxFileName destfile(basepath + dest.Filename);
+			//check that the name is valid for a new file
+			wxString errMsg;
+			if( !isValidNewFilename( newFilename, basepath, errMsg, 5 ) )
+			{
+				wxString message;
+				message.Printf(_("Failed: %s"), WX_STR(errMsg));
+				Msgbox::Alert( message, _("Duplicate memory card") );
+				continue;
+			}
 
-	ScopedBusyCursor doh(Cursor_ReallyBusy);
+			dest.Filename = newFilename.ToStdString();
+			break;
+		}
```

Comment:
This looks really wrong. It looks like you've copy pasted part of this loop inside itself.